src/simulation/logging.rs

```diff
@@ -1,28 +1,44 @@
-use flexi_logger::{detailed_format, Duplicate, FileSpec, Logger, LoggerHandle, WriteMode};
-use log::{info, warn};
+use std::io;
+use tracing::level_filters::LevelFilter;
+use tracing::Level;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_appender::{non_blocking, rolling};
+use tracing_subscriber::fmt;
+use tracing_subscriber::fmt::writer::MakeWriterExt;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::Layer;
 
-pub fn init_logging(directory: &str, file_discriminant: Option<String>) -> Option<LoggerHandle> {
-    let result = Logger::try_with_str("info")
-        .unwrap()
-        .log_to_file(
-            FileSpec::default()
-                .o_discriminant(file_discriminant)
-                .suppress_timestamp()
-                .directory(directory),
-        )
-        .format(detailed_format)
-        .write_mode(WriteMode::Direct)
-        .duplicate_to_stdout(Duplicate::All)
-        .start();
+pub fn init_logging(directory: &str, file_discriminant: String) -> (WorkerGuard, WorkerGuard) {
+    let mut file_name = String::from("mpi_qsim_");
+    file_name.push_str(file_discriminant.as_str());
+
+    let log_file_appender = rolling::never(directory, &file_name);
+    let (log_file, _guard_log) = non_blocking(log_file_appender);
```

Comment:
Do we want non blocking here? This will probably create a writer thread, but threading is already controlled by mpi. Don't know whether this gives an advantage.

Reply:
I wasn't sure either. Non-blocking sounded more performance-friendly to me especially with respect to the logging files. My suggestion: I'm going to merge this PR and during my performance tests, I will compare blocking and non-blocking logging. That shouldn't be too elaborate.